src/bio.c

```diff
@@ -793,6 +837,30 @@ long wolfSSL_BIO_ctrl(WOLFSSL_BIO *bio, int cmd, long larg, void *parg)
         case BIO_CTRL_RESET:
             ret = (long)wolfSSL_BIO_reset(bio);
             break;
+
+#ifdef WOLFSSL_HAVE_BIO_ADDR
+        case BIO_CTRL_DGRAM_CONNECT:
+        case BIO_CTRL_DGRAM_SET_PEER:
+            if (bio->peer_addr)
+                wolfSSL_BIO_ADDR_free(bio->peer_addr);
+            bio->peer_addr = (WOLFSSL_BIO_ADDR *)parg;
```

Comment:
OpenSSL does not take ownership of `parg`. See `BIO_ADDR_make`.

Reply:
Oh dear, yes this was an oversight.  I've refactored `WOLFSSL_BIO.peer_addr` to be inline rather than a pointer, and added the needed copy-ins in place of the pointer assignments.

Thanks for catching that.